news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+
+However, refactoring can only get you so far until it starts to require introducing breaking changes. And once you hit that brick wall, you're stuck: either you let that part of the code rot or you introduce breaking changes that will undermine the adoption of your platform. Or worse, you introduce an _new_ subsystem that does the same thing as the old one, and keep the previous one for compatibility, consequently increasing the overall software complexity.
+
+That's one of the top reasons PrestaShop's architecture has become so complex. The constant need to provide as much backwards compatibility as possible, even when doing so was detrimental to the platform's overall technical quality, has substantially undermined the project's capacity to move forward.
+
+That said, let's find out what other problems are hiding under the hood...
+
+## Plan
+
+* [An aging architecture](#an-aging-architecture)
+* [No clearly-defined API](#no-clearly-defined-api)
+* [1.7 was released too soon](#17-was-released-too-soon)
+* [Too many dependencies with external services](#too-many-dependencies-with external-services)
+* [Testing is very hard](#testing-is-very-hard)
+* [Other problems](#other-problems)
+
+## An aging architecture
+
+PrestaShop has deeply-rooted design issues that can be explained simply by the age of its architecture.
+
+Web technologies have evolved massively in the last years, and industry standards are now very different compared to what they were 10 years ago, when the current foundations of PrestaShop were laid out. Back then, web applications were still being built using the classic backend-generated, form-centric approach that became commonplace during the [CGI-based era](https://en.wikipedia.org/wiki/Common_Gateway_Interface), which forged the modern Internet.
+
+Fast forward to 2019, the web is a very different place. It's a mobile-first world, where experiences provided by [Progressive Web Apps](https://developers.google.com/web/progressive-web-apps/) have become so rich that the frontier between a web site and a native phone app is now at its thinnest. A world where interacting with a form while offline, infinite scroll and instant interaction is now normal, and reloading a page is archaic. A world made of rich front-end applications, and simple API-based backends.
+
+Let's face it, PrestaShop's architecture, as we know it, shows its age. PrestaShop was designed for [a world of Smarty and jQuery][legacy-controllers], but we currently live in a world of React (or Vue, or Angular) and GraphQL.
```

Comment:
You're sure you want to quote GraphQL ? Some people might think it's an endorsement 😝. Maybe stick with REST ?

Reply:
I'm only speaking about hype, popular technologies. Why do you think it could be seen as an endorsement?

Reply:
Because you mention it and you are the lead developer.

Nothing in the sentence looks like an endorsement but ... you know some people love to jump on conclusions 😛

Reply:
Oh I think you're worrying too much 😛